pyquil/api/quantum_computer.py

```diff
@@ -0,0 +1,183 @@
+from math import pi
+from typing import List
+
+import numpy as np
+
+from pyquil.api._qam import QAM
+from pyquil.device import AbstractDevice
+from pyquil.gates import MEASURE, RX
+from pyquil.quil import Program, get_classical_addresses_from_program
+from pyquil.quilbase import Measurement, Pragma
+
+
+def _get_flipped_protoquil_program(program: Program):
+    """For symmetrization, generate a program where X gates are added before measurement.
+
+    Forest 1.3 is really picky about where the measure instructions happen. It has to be
+    at the end!
+    """
+    program = Program(program.instructions)  # Copy
+    to_measure = []
+    while True:
```

Comment:
This won't work for empty programs or programs with only measure instructions (maybe you're okay with this)

Reply:
good catch. fixed in 64f40391d02ede8b4c6972a0000dc1835869afcb